### PR TITLE
update the translation of Array.length

### PR DESCRIPTION
**Pull request by robert-jx**
### Description

更新 array.length 翻译
### Motivation

同上
### Additional details

无

### CODE REVIEW

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 1]
```diff
@@ -5,13 +5,19 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/length
 
 {{JSRef}}
 
-**`length`** 是 `Array` 的实例属性。返回或设置一个数组中的元素个数。该值是一个无符号 32-bit 整数，并且总是大于数组最高项的下标。
+**`length`** 是 `Array` 的实例属性表示该数组中元素的个数。该值是一个无符号 32-bit 整数，并且其数值总是大于数组最高项的下标。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:8]:
```suggestion
**`length`** 是 `Array` 的实例属性，表示该数组中元素的个数。该值是一个无符号 32 位整数，并且其数值总是大于数组最大下标。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 2]
```diff
@@ -5,13 +5,19 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/length
...
 
-{{EmbedInteractiveExample("pages/js/array-length.html")}}
+{{EmbedInteractiveExample("pages/js/array-length.html","shorter")}}
+
+## 值
+
+值是一个小于 2<sup>32</sup> 的非负整数。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:14]:
```suggestion
一个小于 2<sup>32</sup> 的非负整数。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 3]
```diff
@@ -5,13 +5,19 @@ slug: Web/JavaScript/Reference/Global_Objects/Array/length
...
+
+{{js_property_attributes(1, 0, 0)}}
 
 ## 描述
 
-`length` 属性的值是一个 0 到 2^32 - 1 的整数。
+`length` 属性的值是一个非负整数，值小于 2<sup>32</sup>。
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:20]:
```suggestion
`length` 属性的值是一个小于 2<sup>32</sup> 的非负整数
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 4]
```diff
@@ -46,32 +58,26 @@ arr.forEach(element => console.log(element));
...
-
-- `Writable`：如果设置为 `false`，该属性值将不能被修改。
-- `Configurable`：如果设置为 `false`，删除或更改任何属性都将会失败。
-- `Enumerable`：如果设置为 `true`，属性可以通过迭代器[for](zh-CN/docs/Web/JavaScript/Reference/Statements/for)或[for...in](/zh-CN/docs/Web/JavaScript/Reference/Statements/for...in)进行迭代。
+另请参阅：[`length`与数值属性的关系](/zh-CN/docs/Web/JavaScript/Reference/Global_Objects/Array#长度与数值属性的关系)。
 
 ## 示例
 
-### 遍历数组
+### 在数组上迭代
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:65]:
```suggestion
### 遍历数组
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 5]
```diff
@@ -46,32 +58,26 @@ arr.forEach(element => console.log(element));
...
 
 ```js
 const numbers = [1, 2, 3, 4, 5];
 const length = numbers.length;
-for (var i = 0; i < length; i++) {
+for (let i = 0; i < length; i++) {
   numbers[i] *= 2;
 }
-// 遍历后的结果 [2, 4, 6, 8, 10]
+// numbers is now [2, 4, 6, 8, 10]
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:75]:
```suggestion
// 遍历后的结果 [2, 4, 6, 8, 10]
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 6]
```diff
@@ -82,6 +88,30 @@ if (numbers.length > 3) {
 
 console.log(numbers); // [1, 2, 3]
 console.log(numbers.length); // 3
+console.log(numbers[3]); // undefined; the extra elements are deleted
```
Comment (JasonLamv-t) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:91]:
```suggestion
console.log(numbers[3]); // undefined；多余的元素会被删除
```

**Review by JasonLamv-t — COMMENTED** [review]
Some suggestions.

**Comment by JasonLamv-t** [comment]
Some problem in https://github.com/mdn/content/pull/26339

**Comment by JasonLamv-t** [comment]
https://github.com/mdn/content/pull/26339 already determined. Please modify the title meta according to the original text.

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 7]
```diff
@@ -1,17 +1,23 @@
 ---
-title: Array.length
+title: "Array: length"
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:2]:
```suggestion
title: Array：length
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 8]
```diff
@@ -1,17 +1,23 @@
 ---
-title: Array.length
+title: Array：length
 slug: Web/JavaScript/Reference/Global_Objects/Array/length
 ---
 
 {{JSRef}}
 
-**`length`** 是 `Array` 的实例属性。返回或设置一个数组中的元素个数。该值是一个无符号 32-bit 整数，并且总是大于数组最高项的下标。
+**`length`** 是 `Array` 的实例属性，表示该数组中元素的个数。该值是一个无符号 32 位整数，并且其数值总是大于数组最大下标。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:8]:
```suggestion
**`length`** 是 `Array` 的实例属性，表示该数组中元素的个数。该值是一个无符号 32 位整数，并且其数值总是大于数组最大索引。
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 9]
```diff
@@ -23,21 +29,27 @@ console.log(listA.length);
 console.log(listB.length);
 // 6
 
-listB.length = 4294967296; // 2 的 32 次方 = 4294967296
+listB.length = 2 ** 32; // 2 的 32 次方 = 4294967296
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:32]:
```suggestion
listB.length = 2 ** 32; // 4294967296
```

File: files/zh-cn/web/javascript/reference/global_objects/array/length/index.md [hunk 10]
```diff
@@ -82,6 +88,30 @@ if (numbers.length > 3) {
...
+
+```js
+const numbers = [];
+numbers.length = 3;
+console.log(numbers); // [empty x 3]
+```
+
+### 长度不可写的数组
+
+当添加的元素超过当前长度时，数组会自动更新 `length` 属性。如果 `length` 属性设置为不可写，则数组将无法更新它。这会导致[严格模式](/zh-CN/docs/Web/JavaScript/Reference/Strict_mode)错误。
```
Comment (yin1999) [inline files/zh-cn/web/javascript/reference/global_objects/array/length/index.md:106]:
```suggestion
当添加的元素超过当前长度时，数组会自动更新 `length` 属性。如果 `length` 属性设置为不可写，则数组将无法更新它。在[严格模式](/zh-CN/docs/Web/JavaScript/Reference/Strict_mode)中这会导致错误。
```

**Review by yin1999 — APPROVED** [review]

**Pull request merged**